packages/p2p/src/hooks/__tests__/use-p2p-rendered-adverts.spec.ts

```diff
@@ -0,0 +1,331 @@
+        const view = result.current.rendered_adverts;
+
+        expect(view).toHaveLength(1);
+        expect(view[0]?.country).toBe('id');
+        expect(view[0].counterparty_type).toBe('buy');
+        expect(view[0].description).toBe('Created by script. Please call me 02203400');
+        expect(view[0].advertiser_details?.name).toBe('client Test90000253');
+    });
+    it('should return the adverts object from response when selected_local_currency is defined', () => {
+        mockUseStores.mockReturnValue({
+            general_store: {
+                lisst_item_limit: 50,
+            },
+            buy_sell_store: {
+                selected_local_currency: undefined,
+                selected_payment_method_value: 'USD',
+                table_type: 'sell',
+            },
+            advertiser_page_store: { counterparty_type: 'buy' },
+        });
+        const { result } = renderHook(() => useP2PRenderedAdverts());
+        const view = result.current.rendered_adverts;
+
+        expect(view).toHaveLength(1);
+        expect(view[0]?.country).toBe('id');
+        expect(view[0].counterparty_type).toBe('buy');
+        expect(view[0].description).toBe('Created by script. Please call me 02203400');
+        expect(view[0].advertiser_details?.name).toBe('client Test90000253');
+    });
+    it('should return the adverts object from response when selected_local_currency is undefined', () => {
+        mockUseStores.mockReturnValue({
+            general_store: {
+                lisst_item_limit: 50,
+            },
+            buy_sell_store: {
+                selected_local_currency: undefined,
+                selected_payment_method_value: '',
+                table_type: 'sell',
+            },
+            advertiser_page_store: { counterparty_type: 'buy' },
+        });
+        const { result } = renderHook(() => useP2PRenderedAdverts());
+        const view = result.current.rendered_adverts;
+
+        expect(view).toHaveLength(1);
+        expect(view[0]?.country).toBe('id');
+        expect(view[0].counterparty_type).toBe('buy');
+        expect(view[0].description).toBe('Created by script. Please call me 02203400');
+        expect(view[0].advertiser_details?.name).toBe('client Test90000253');
+    });
+    it('should return the adverts object from response when should_use_client_limits is true', () => {
+        mockUseStores.mockReturnValue({
+            general_store: {
+                lisst_item_limit: 50,
+            },
+            buy_sell_store: {
+                selected_local_currency: undefined,
+                selected_payment_method_value: '',
+                table_type: 'sell',
+                should_use_client_limits: true,
+            },
+            advertiser_page_store: { counterparty_type: 'buy' },
+        });
+        const { result } = renderHook(() => useP2PRenderedAdverts());
+        const view = result.current.rendered_adverts;
+
+        expect(view).toHaveLength(1);
```

Comment:
I see this block of expectations are repeated. Can we move it to separate function and reuse? :)